core/shared/src/test/scala/cats/parse/ParserTest.scala

```diff
@@ -562,6 +572,13 @@ class ParserTest extends munit.ScalaCheckSuite {
     parseTest(Parser.product(fooP, barP), "foobar", ((), ()))
   }
 
+  test("longest match stringIn1") {
+    parseTest(Parser.stringIn1(List("foo", "foobar", "foofoo", "foobat")).string, "foo", "foo")
+    parseTest(Parser.stringIn1(List("foo", "foobar", "foofoo", "foobat")).string, "foobat", "foobat")
+    parseTest(Parser.stringIn1(List("foo", "foobar", "foofoo", "foobat")).string, "foot", "foo")
+    parseTest(Parser.stringIn1(List("foo", "foobar", "foofoo", "foobat")).string, "foobal", "foo")
+  }
+
```

Comment:
can we do a law such as "if oneOf matches with all string1 parsers, then stringIn1 would match"

something like:
```scala
forAll { (ss0: List[String], toParse: String) =>
  val ss = ss0.filterNot(_.isEmpty)
  val oneOfs = P.oneOf1(ss.map(P.string1))
  val stringIn = P.stringIn1(ss)
  if (oneOfs.parse(toParse).isRight) assert(stringIn.parse(toParse).isRight)
}
```
or something like that.

I'd like to find more laws like that so we can stress it a bit harder.

Also a law that stringIn1(s :: Nil) == string1(s)
etc...

Reply:
Both would match but parsed result might differ, do you think this is a strong enough law?

Reply:
any law is better than no law, which is the current state. Let's add this, but also any others you can find.

another law might be the direct law:

if any of these would match with string1, then stringIn will match and return the longest match. Something like that?